### PR TITLE
feat(cart): add place order Magento driver

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
- Adds a place order driver
- Adds a bad input error
- Changes `transformError` to use the general version in `@daffodil/driver/magento`

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/cart/src/errors/bad-input.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export class DaffBadInputError extends Error {
```
Comment (lderrickable) [inline libs/cart/src/errors/bad-input.ts:1]: This seems like it should go in core, but I'm not sure. Or maybe it should be named `DaffBadCartInputError`

Comment (griest024) [reply libs/cart/src/errors/bad-input.ts:1]: Yeah you're right. This will prob be used in most of the Magento drivers.

Comment (griest024) [reply libs/cart/src/errors/bad-input.ts:1]: I'll put it in `driver/magento`

File: libs/cart/src/drivers/magento/cart-order.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartOrderService implements DaffCartOrderServiceInterface<DaffCart> {
+  constructor(
+    private apollo: Apollo,
+    public cartTransformer: DaffMagentoCartTransformer,
+  ) {}
+
+  placeOrder(cartId: DaffCart['id']) {
```
Comment (lderrickable) [inline libs/cart/src/drivers/magento/cart-order.service.ts:26]: could use a return type

File: libs/cart/src/errors/bad-input.ts (hunk 1 above)
Comment (griest024) [reply libs/cart/src/errors/bad-input.ts]: done

File: libs/cart/src/drivers/magento/cart-order.service.ts (hunk 2 above)
Comment (griest024) [reply libs/cart/src/drivers/magento/cart-order.service.ts]: fixed

File: libs/cart/src/errors/bad-input.ts (hunk 1 above)
Comment (griest024) [reply libs/cart/src/errors/bad-input.ts]: I also changed the transform error function to use the one in `driver`.

File: libs/driver/magento/src/errors/public_api.ts [hunk 3]
```diff
@@ -1 +1,2 @@
 export { daffTransformMagentoError } from './transform';
+export { DaffBadInputError } from './bad-input';
```
Comment (lderrickable) [inline libs/driver/magento/src/errors/public_api.ts:2]: Seems to me like the `daffTransformMagentoError` should be here is `@daffodil/driver/magento`, but the actual `DaffBadInputError` should be in `@daffodil/driver`

Comment (griest024) [reply libs/driver/magento/src/errors/public_api.ts:2]: hm that's a good point, it does seem like it'll be useful for all platform drivers.

Comment (griest024) [reply libs/driver/magento/src/errors/public_api.ts]: moved it

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**